make/riot/riot.mk

```diff
@@ -1,12 +1,31 @@
 RIOT_MK_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
 
-# Include generated sources and makefiles if LF_SRC_GEN_PATH is defined
-ifdef LF_SRC_GEN_PATH
-include $(LF_SRC_GEN_PATH)/Makefile
+# Include generated sources and makefiles if LF_MAIN is defined
+ifdef LF_MAIN
+# Execute the LF compiler
+.PHONY: lfcg
+lfcg:
+	$(REACTOR_UC_PATH)/lfc/bin/lfc-dev src/$(LF_MAIN).lf
+
+all: lfcg
+
+# Check if REACTOR_UC_PATH variable exist
+ifndef REACTOR_UC_PATH
+$(error REACTOR_UC_PATH is not defined. Please define it!)
+endif
 
 # Name of your RIOT application
 APPLICATION ?= $(LF_MAIN)
 
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/RIOT
```

Comment:
This assumes that you are building your application from our lf-riot-template and have RIOT as a submodule. Ideally we would keep such assumptions in the makefile within `lf-riot-template`. IIRC we discussed moving some of this back to the makefile in `lf-riot-template`, did you request my review before or after the meeting :stuck_out_tongue:

Reply:
Yeah that was before the meeting haha :)

Reply:
Done